src/librustc_typeck/check/pat.rs

```diff
@@ -705,6 +734,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 subpats.len(),
             ))
             .span_label(res_span, format!("{} defined here", res.descr()))
+            .span_label(pat_span, format!("{}", missing_paranthesis_error))
```

Comment:
Two points here:
- We should use https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/struct.DiagnosticBuilder.html#method.span_suggestion instead of `.span_label` to add a `MachineApplicable` suggestion `(pat0, pat1, ...)`

   - You can use `let mut err = ...;` since chaining won't work well in this case anymore sadly.
   - The `suggestion: String` will need to be built out of the stringified patterns (use https://doc.rust-lang.org/nightly/nightly-rustc/syntax/print/pprust/fn.pat_to_string.html), interspersed with `,` and then surrounded with `(` and `)`.

- The `pat_span` here could be replaced with `subpats[0].span.to(subpats.last().unwrap().span)` to give a more targeted `Span`. This will be useful for the suggestion aforementioned.

cc @estebank in case he has thoughts.

Reply:
`Span`s have `shrink_to_lo`/`shrink_to_hi` methods that point at the space right before the first span and right after the last span, respectively. You can then make a `multispan_suggestion` with little logic:

```rust
err.multispan_suggestion(
    vec![(pat_span.shrink_to_lo(), "(".to_string()), (pat_span.shrink_to_hi(), ")".to_string())],
    &missing_parenthesis_error,
    Applicability::MachineApplicable,
);
```

Reply:
@estebank There is a `multipart suggestion` module. I think you mean that?

Reply:
Yes, that's the correct name :)